app/helpers/application_helper.rb

```diff
@@ -105,9 +105,9 @@ def role_allows?(**options)
     end
 
     # Detect if queried features are missing from the database and possibly invalid
-    if !Rails.env.production? && MiqProductFeature.where(:identifier => features).count != features.length
+    if !Rails.env.production? && features.detect { |feature| !MiqProductFeature.feature_exists?(feature) }
       message = "#{__method__} no feature was found with identifier: #{features.inspect}.  Correct the identifier or add it to miq_product_features.yml."
-      identifiers = MiqProductFeature.all.pluck(:identifier)
+      identifiers = MiqProductFeature.features.keys
```

Comment:
Does the `MiqProductFeature.features` cached get busted when we have new dynamic features (i.e. those weird ones when you create a new tenant and thus they are tenant specific)?  If so, then this is awesome!

Reply:
Yes, they are invalidated [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/tenant.rb#L65) (I was worried about that for a minute

We really need to get rid of those. Product features per tenant was not the best idea we've had.

Reply:
Agreed - not sure how we undo that at the moment though.